server/src/server.js

```diff
@@ -49,6 +49,10 @@ app.use(function (req, res, next) {
 app.use(function (req, res, next) {
   req.staticLink = linker.staticLink;
   req.staticLinkWithHost = linker.staticLinkWithHost.bind(null, req);
+  if (Shot.prototype.imageLink === undefined) {
```

Comment:
Not sure why this is going on the Shot prototype?  Also this implicitly means that the first request permanently defines the host for the life of the process.